railties/lib/rails/command.rb

```diff
@@ -92,6 +93,10 @@ def print_commands # :nodoc:
         COMMANDS_IN_USAGE = %w(generate console server test test:system dbconsole new)
         private_constant :COMMANDS_IN_USAGE
 
+        def rails_new_with_no_path?(args, config)
```

Comment:
I couldn't find a way in `thor` to have behavior run if a required argument is missing. It just [aborts](https://github.com/erikhuda/thor/blob/011dc48b5ea92767445b062f971664235973c8b4/lib/thor/parser/arguments.rb#L176).

So instead I chose to handle this before we even reach the `thor` layer.